Tests/test_SeqIO_StructureIO.py

```diff
@@ -0,0 +1,185 @@
+    from numpy.linalg import svd, det  # Missing in PyPy 2.0 numpypy
+except ImportError:
+    from Bio import MissingPythonDependencyError
+    raise MissingPythonDependencyError(
+        "Install NumPy if you want to use PDB formats with SeqIO.")
+
+from Bio import SeqIO
+from Bio import BiopythonWarning
+from Bio.PDB.PDBExceptions import PDBConstructionWarning
+
+
+def SeqresTestGenerator(extension, parser):
+    """
+    Test factory for tests reading SEQRES (or similar) records.
+
+    This is a factory returning a parameterised superclass for tests reading
+    sequences from the sequence records of structure files.
+
+    Arguments:
+        extension:
+            The extension of the files to read from the ``PDB`` directory (e.g.
+            ``pdb`` or ``cif``).
+        parser:
+            The name of the SeqIO parser to use (e.g. ``pdb-atom``).
+    """
+
+    class SeqresTests(unittest.TestCase):
+        """Use "parser" to parse sequence records from a structure file.
+
+        Args:
+            parser (str): Name of the parser used by SeqIO.
+            extension (str): Extension of the files to parse.
+        """
+
+        def test_seqres_parse(self):
+            """Parse a multi-chain PDB by SEQRES entries.
+
+            Reference:
+            http://www.rcsb.org/pdb/files/fasta.txt?structureIdList=2BEG
+            """
+            chains = list(SeqIO.parse('PDB/2BEG.' + extension, parser))
+            self.assertEqual(len(chains), 5)
+            actual_seq = 'DAEFRHDSGYEVHHQKLVFFAEDVGSNKGAIIGLMVGGVVIA'
+            for chain, chn_id in zip(chains, 'ABCDE'):
+                self.assertEqual(chain.id, '2BEG:' + chn_id)
+                self.assertEqual(chain.annotations['chain'], chn_id)
+                self.assertEqual(str(chain.seq), actual_seq)
+
+        def test_seqres_read(self):
+            """Read a single-chain structure by sequence entries.
+
+            Reference:
+            http://www.rcsb.org/pdb/files/fasta.txt?structureIdList=1A8O
+            """
+            chain = SeqIO.read('PDB/1A8O.' + extension, parser)
+            self.assertEqual(chain.id, '1A8O:A')
+            self.assertEqual(chain.annotations['chain'], 'A')
+            self.assertEqual(str(chain.seq),
+                             'MDIRQGPKEPFRDYVDRFYKTLRAEQASQEVKNWMTETLLVQNANPD'
+                             'CKTILKALGPGATLEEMMTACQG')
+
+        def test_seqres_missing(self):
+            """Parse a PDB with no SEQRES entries."""
+            chains = list(SeqIO.parse('PDB/a_structure.' + extension, parser))
+            self.assertEqual(len(chains), 0)
+    return SeqresTests
+
+
+class TestPdbSeqres(SeqresTestGenerator("pdb", "pdb-seqres")):
+    """Test pdb-seqres SeqIO driver."""
+    pass
+
+
+class TestCifSeqres(SeqresTestGenerator("cif", "cif-seqres")):
+    """Test cif-seqres SeqIO driver."""
+    pass
+
+
+def AtomTestGenerator(extension, parser):
+    """
+    Test factory for tests reading ATOM (or similar) records.
```

Comment:
Please remove the new line to follow PEP257 here
https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings